src/main/java/ch/njol/skript/expressions/ExprName.java

```diff
@@ -132,6 +133,10 @@ public class ExprName extends SimplePropertyExpression<Object, String> {
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		mark = parseResult.mark;
 		setExpr(exprs[0]);
+		if (mark != 1 && World.class.isAssignableFrom(getExpr().getReturnType())) {
+			Skript.error("Use 'name' to get the name of a world.");
```

Comment:
Better error message
```suggestion
			Skript.error("Can't use 'display name' with worlds. Use 'name' instead");
```